File: problems/10 - Summation of primes.py
'''The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
Find the sum of all the primes below two million.'''


k = 2000000
marker = [True] * k

for num in range(2, round(k**0.5)):
    if marker[num]:
        for multiple in range(num + num, k, num):
            marker[multiple] = False

primes = list(range(2, k))
print(sum([prime for prime in primes if marker[prime]]))

File: c solutions/10_summation_of_primes.c
#include <stdio.h>
#include <math.h>


int main() {
    int k = 2000000;
    char primes[k + 1];
    for (int i = 0; i < k; i++) {
        primes[i] = 1;
    }

    for (int range = 2; range <= (ceil(sqrt(k))); range++) {
        if (primes[range] == 1) {
            for (int multiple = range * 2; multiple <= k; multiple += range) {
                primes[multiple] = 0;
            }
        }
    }
    
    long long sum = 0;
    for (int i = 2; i <= k; i++) {
        if (primes[i] == 1) {
            sum += i;
        }
    }

    printf("%lld", sum);
    return 0;
}